components/inputs/input-number.js

```diff
@@ -0,0 +1,83 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { inputLabelStyles } from './input-label-styles.js';
+import { inputStyles } from './input-styles.js';
+import { offscreenStyles } from '../offscreen/offscreen.js';
+
+class InputNumber extends LitElement {
+
+	static get properties() {
+		return {
+			autocomplete: { type: String },
+			autofocus: { type: Boolean },
+			disabled: { type: Boolean },
+			label: { type: String },
+			labelHidden: { type: Boolean, attribute: 'label-hidden' },
+			max: { type: Number },
+			min: { type: Number },
+			name: { type: String },
+			placeholder: { type: String },
+			required: { type: Boolean },
+			step: { type: Number },
+			value: { type: Number },
+			width: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [
+			inputLabelStyles,
+			inputStyles,
+			offscreenStyles,
+			css`
+				:host {
+					display: inline-block;
+					width: 4rem;
+				}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this.autofocus = false;
+		this.disabled = false;
+		this.labelHidden = false;
+		this.required = false;
+
+		this._inputId = getUniqueId();
+	}
+
+	render() {
+		const ariaRequired = this.required ? 'true' : undefined;
+
+		if (ifDefined(this.width)) {
+			this.style.width = this.width;
+		}
+
+		return html`
+			<label
+				class="${this.label && !this.labelHidden ? 'd2l-input-label' : 'd2l-offscreen'}"
+				for="${this._inputId}"
+				id="${this._inputId}-label">${this.label}</label>
+			<input
+				aria-labelledby="${this._inputId}-label"
```

Comment:
The `for` attribute on the label links these two together -- no need for `aria-labelledby` on the input (or the `id` on the label).